bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/tracer/BSpan.java

```diff
@@ -178,19 +176,15 @@ public BMap<BString, Object> getBSpanContext() {
 
         if (bSpanContext == null) {
             SpanContext spanContext = span.getSpanContext();
-            if (spanContext.isSampled()) {
-                BMapInitialValueEntry[] values = new BMapInitialValueEntry[]{
-                        new MappingInitialValueEntry.KeyValueEntry(
-                                TraceConstants.SPAN_CONTEXT_MAP_KEY_TRACE_ID,
-                                StringUtils.fromString(spanContext.getTraceId())),
-                        new MappingInitialValueEntry.KeyValueEntry(
-                                TraceConstants.SPAN_CONTEXT_MAP_KEY_SPAN_ID,
-                                StringUtils.fromString(spanContext.getSpanId()))
-                };
-                bSpanContext = ValueCreator.createMapValue(IMMUTABLE_STRING_MAP_TYPE, values);
-            } else {
-                bSpanContext = EMPTY_BSPAN_CONTEXT;
-            }
+            BMapInitialValueEntry[] values = new BMapInitialValueEntry[]{
+                    new MappingInitialValueEntry.KeyValueEntry(
+                            TraceConstants.SPAN_CONTEXT_MAP_KEY_TRACE_ID,
+                            StringUtils.fromString(spanContext.getTraceId())),
+                    new MappingInitialValueEntry.KeyValueEntry(
+                            TraceConstants.SPAN_CONTEXT_MAP_KEY_SPAN_ID,
+                            StringUtils.fromString(spanContext.getSpanId()))
+            };
+            bSpanContext = ValueCreator.createMapValue(IMMUTABLE_STRING_MAP_TYPE, values);
```

Comment:
Can we directly return map value?

Reply:
No. `bSpanContext` is an instance variable. We are store it in BSpan object and reuse it for subsequent calls.